app/controllers/application_controller.rb

```diff
@@ -8,6 +8,7 @@ def login_required
   end
 
   def set_user
+    current_user.authentication_token = nil if current_user
```

Comment:
hm I don't think this should be the way to do it.  If someone goes to the edit page and save. They lost their token. 
Instead, you should change the as_json function like the follows: 

 def to_json(options={})
     options[:except] ||= [:encrypted_password, :authentication_token, etc ]
     super(options)
   end